### PR TITLE
Fix Agent crash on shutdown

### PR DESCRIPTION
**Pull request by Atlante45**
[Manuscript ticket](https://highfidelity.manuscript.com/f/cases/20081/Agent-shut-down-crash-while-taking-down-the-ClientTraitsHandler)

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15509/HighFidelity-Beta-PR14494-9127a2a.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15509/HighFidelity-Beta-PR14494-9127a2a.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15509/HighFidelity-Beta-Interface-PR14494-9127a2a.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15509/HighFidelity-Beta-PR14494-9127a2a.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15509/HighFidelity-Beta-Interface-PR14494-9127a2a.exe

File: libraries/shared/src/SharedUtil.h [hunk 1]
```diff
@@ -113,6 +113,13 @@ void doEvery(quint64& lastReportUsecs, quint64 secs, F lamdba) {
 // Maximum accuracy in msecs
 float secTimestampNow();
 
+// Custom deleter for QObjects that calls deleteLater
+struct LaterDeteter {
```
Comment (kencooke) [inline libraries/shared/src/SharedUtil.h]: typo, should be LaterDeleter

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15512/HighFidelity-Beta-PR14494-181f2b4.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15512/HighFidelity-Beta-Interface-PR14494-181f2b4.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15512/HighFidelity-Beta-PR14494-181f2b4.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15512/HighFidelity-Beta-Interface-PR14494-181f2b4.exe

**Comment by antoninahifi** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15586/HighFidelity-Beta-PR14494-181f2b4.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15586/HighFidelity-Beta-PR14494-181f2b4.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15586/HighFidelity-Beta-Interface-PR14494-181f2b4.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15586/HighFidelity-Beta-PR14494-181f2b4.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15586/HighFidelity-Beta-Interface-PR14494-181f2b4.exe

**Comment by antoninahifi** [comment]
currently testing

**Comment by antoninahifi** [comment]
@conklin94122 PASS

**Pull request merged**